contracts/IPAccountImpl.sol

```diff
@@ -56,42 +55,30 @@ contract IPAccountImpl is IPAccountStorage, IIPAccount {
     /// @return chainId The EIP-155 ID of the chain the token exists on
     /// @return tokenContract The contract address of the token
     /// @return tokenId The ID of the token
-    function token() public view override returns (uint256, address, uint256) {
-        bytes memory footer = new bytes(0x60);
-        // 0x4d = 77 bytes (ERC-1167 Header, address, ERC-1167 Footer, salt)
-        // 0x60 = 96 bytes (chainId, tokenContract, tokenId)
-        //    ERC-1167 Header               (10 bytes)
-        //    <implementation (address)>    (20 bytes)
-        //    ERC-1167 Footer               (15 bytes)
-        //    <salt (uint256)>              (32 bytes)
-        //    <chainId (uint256)>           (32 bytes)
-        //    <tokenContract (address)>     (32 bytes)
-        //    <tokenId (uint256)>           (32 bytes)
-        assembly {
-            extcodecopy(address(), add(footer, 0x20), 0x4d, 0x60)
-        }
-
-        return abi.decode(footer, (uint256, address, uint256));
+    function token() public view override(ERC6551, IIPAccount) returns (uint256, address, uint256) {
+        return super.token();
     }
 
     /// @notice Checks if the signer is valid for the given data
     /// @param signer The signer to check
     /// @param data The data to check against
     /// @return The function selector if the signer is valid, 0 otherwise
-    function isValidSigner(address signer, bytes calldata data) external view returns (bytes4) {
-        if (_isValidSigner(signer, address(0), data)) {
-            return IERC6551Account.isValidSigner.selector;
-        }
-
-        return bytes4(0);
+    function isValidSigner(
+        address signer,
+        bytes calldata data
+    ) public view override(ERC6551, IIPAccount) returns (bytes4) {
+        return super.isValidSigner(signer, data);
     }
 
     /// @notice Returns the owner of the IP Account.
     /// @return The address of the owner.
-    function owner() public view returns (address) {
-        (uint256 chainId, address contractAddress, uint256 tokenId) = token();
-        if (chainId != block.chainid) return address(0);
```

Comment:
we don't need to do chainId checking?

Reply:
Yes, we need this check. 
This check has been covered in Solady 6551's owner() function. as IPAccount inherit from Solady 6551 now.
we delegate the owner() to solady at Line 76  ` return super.owner();`